### PR TITLE
add custom git url

### PR DESCRIPTION
**Pull request by JackZhouCn**

### CODE REVIEW

**Comment by devxoul** [comment]
@JackZhouCn, Thank you for your work!
How about using `git` instead of `customgit`? I think it's more reasonable.

File: lib/cocoaseeds/core.rb [hunk 1]
```diff
@@ -315,6 +315,53 @@ def bitbucket(repo, tag, options={})
...
 
+    # Creates a new instance of {#Seeds::Seed::BitBucket} and adds to
+    # {#seeds}.
+    #
+    # @see #Seeds::Seed::CustomGit
+    #
+    # @!scope method
+    # @!visibility private
+    #
+    def customgit( repo, tag, options={})
```
Comment (devxoul) [inline lib/cocoaseeds/core.rb]: Unnecessary space after parenthesis.

File: lib/cocoaseeds/core.rb [hunk 2]
```diff
@@ -315,6 +315,53 @@ def bitbucket(repo, tag, options={})
...
+            elsif repo.split('/').count != 2
+            raise Seeds::Exception.new\
+            "#{repo}: GitHub should have both username and repo name.\n"\
+            "    (e.g. `devxoul/JLToast`)"
+            else
+                seed = Seeds::Seed::GitHub.new
+                seed.url = "#{repo}"
+                seed.name = repo.split('/')[1]
+                if tag.is_a?(String)
+                if options[:commit]
```
Comment (devxoul) [inline lib/cocoaseeds/core.rb]: I think 341~347 line should be indented one step.

**Comment by JackZhouCn** [comment]
I make some improvements, thank you.

**Comment by devxoul** [comment]
@JackZhouCn,

I think `git` function should be more careful. For example, it should remove `.git` suffix from url automatically, and check validate git urls strictly.

This is not good: `Installing JLToast.git (1.3.0)`

And `git@github.com/JLToast.git` url should make an exception.

I think `git` function can be the replacement for `github` and `bitbucket` function.

**Comment by ryanholden8** [comment]
Hi @devxoul, any update on this PR? Would love to use this project but need to be able to import internal git resources. Thank you!

**Comment by veightz** [comment]
Aha! It's what I want!
In fact, the syntax I created like this:

```
source :PrivateGitSource0, "git@gitlab.example.com"
source :PrivateGitSource1, "git@gitlab.example.com"

target :SomeTarget do
  # seed is a method used for custom git repo.
  seed "somegroup0/somerepo0", "master", :from=>:PrivateGitRepo0, :files => "xxxx/**/*.{h,m,mm,swift,plist,png}"
  seed "somegroup1/somerepo1", "master", :from=>:PrivateGitRepo1, :files => "yyyy/**/*.{h,m,mm,swift,plist,png}"

  # develop is a method used for local git repo to test some feature.
  develop "zzzz", "../zzzz", :files => "zzzz/**/*.{h,m,mm,swift,plist,png}"  
end
```

**Comment by devxoul** [comment]
@veightz, I think git should be `git` and local files should be `local`.

**Comment by devxoul** [comment]
duplicated: #58

**Pull request closed without merging**